[ADD] Add option for Odoo 17.0

celm1990 opened the pull request:
I'm not sure if it's a good idea to use Copier 9, but I have problems with the old version. @yajo, can you please review?

TODO: adapt github actions to V17 and PY 3.10

Is need execute` copier copy` with option `--trust ` similar to https://github.com/OCA/oca-addons-repo-template/pull/202

pedrobaeza commented on copier.yml:
```diff
@@ -1,16 +1,8 @@
 # Configure jinja2 defaults to make syntax highlighters lives easier
 _templates_suffix: .jinja
-_envops:
```
Why do you remove this?

celm1990 replied:
see https://github.com/OCA/oca-addons-repo-template/pull/221#discussion_r1377253209

celm1990 commented:
@pedrobaeza @yajo 

Please review the comments associated with each commit for additional details.

I attempted to adapt the test, but encountered errors during precommit. I suspect this is because the oca-addons-repo-template now utilizes ruff instead of black, isort, flake8, autoflake, and pyupgrade. However, I am uncertain whether the maximum line length is now set to 79 instead of 88. @sbidoul, could you please provide your insights on this matter?



Refer to the following commit for the changes:  https://github.com/OCA/oca-addons-repo-template/commit/49b350fd795e039963e153943a8d0635be6f0fee

![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/20102c94-2af9-4b6f-bb04-11760dbaf7e1)

pedrobaeza commented:
Better to do this in several steps: limit this PR to enable the 17.0 version, and then we can investigate about switching to `ruff` in another one. Check anyway the file https://github.com/OCA/oca-addons-repo-template/blob/master/src/%7B%25%20if%20use_ruff%20%25%7D.ruff.toml%7B%25%20endif%25%7D.jinja for ruff options.

celm1990 commented:
> Better to do this in several steps: limit this PR to enable the 17.0 version, and then we can investigate about switching to `ruff` in another one.

Currently, the process is not functional without some adjustments. This is necessary due to changes made in the `oca-addons-repo-template` repository first on [this commit](https://github.com/OCA/oca-addons-repo-template/commit/299cb36409bc774359508f6dc46ccf18ce7290dc) the files `.flake8` `.isort.cfg` were removed. Following that, in
 [this commit](https://github.com/OCA/oca-addons-repo-template/commit/072cfae2e9ae70cfdb62a3aa93e181ef10d7bb4d) an option for version 17 was recently added. As a result, it's imperative to adapt to these changes.

For visual reference, please find the screenshot below:


![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/a1ab77f5-29b3-4abf-93a6-374ca6922ce9)

celm1990 commented:
@pedrobaeza  I updated code for `flake8` and `isort`, before the file content are taken from oca-addons-repo-template but on this PR are delete this files https://github.com/OCA/oca-addons-repo-template/pull/219 So now I put content directly here. Can you enable github actions and review please

yajo commented on tests/samples/mqt-diffs/v10.0-.pylintrc-mandatory.diff:
thought: these diff files are less helpful each time. The origin idea behind these files was to implement some kind of [snapshot testing](https://jestjs.io/docs/snapshot-testing) that served us as a warning sign when we diverged from the recommended linter configurations from OCA.

However, it's impossible for a human to interpret correctly 16 plain diffs in a review.

Back then, when `oca-addons-repo-template` didn't exist, OCA was using standard configurations per version, based on their MQT repo, which is mostly useless today (https://github.com/OCA/maintainer-quality-tools/issues/717).

Now that updates are based on Copier, and don't happen randomly, what's the point on freezing the linter configurations per version? It's useless IMHO. Linters don't happen at Odoo runtime. They happen at dev time, in the dev laptop. Or on CI, in separate jobs. You can lint today code for Odoo 8 using today's linters, with a modern python version, and it will lint it properly.

So linters should depend on the template version, and not on the odoo version. In OCA they don't think like this and it's a maintenance burden in their template. I don't think we should follow the same path.

IMHO we should just review the new linters, enable those that make sense for all supported versions, and remove this diff snapshotting method.

celm1990 replied:
Hi, this is my first contribution into `doodba-copier-template` and I attempted to adhere to [this guide](https://github.com/Tecnativa/doodba-copier-template/blob/main/CONTRIBUTING.md). However, upon executing  `invoke update-test-samples` all diff are altered,  I'm having difficulty understanding why it has become more complex.
Do you recommend removing all the test related?

Currently, my primary objective is to enable V17 for Doodba. Unfortunately, the testing process is proving to be intricate and a bit frustrating. :(

pedrobaeza replied:
Let's remove the diff tests all together then.

celm1990 replied:
I removed all test related, can you enable github actions please .
On my repo get the followin error(The test attempts to commit with a manifest containing a higher version, and it should fail the pre-commit check) , but Pylint-odoo working properly up to version v8.0.5, but starting from version v8.0.6, it encounters issues., please check issue reported [here](https://github.com/OCA/oca-addons-repo-template/issues/230#issuecomment-1807218563)
![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/e7625722-db1c-4daa-9032-ad6155d0f22c)

pedrobaeza replied:
AFAIK, the names of the linters to execute changed on that version from underscores (`_`) to dashes (`-`).

celm1990 replied:
> AFAIK, the names of the linters to execute changed on that version from underscores (`_`) to dashes (`-`).

The configuration is OK, issue become from pylint-odoo not support V17.0 yet.
waiting https://github.com/OCA/pylint-odoo/pull/477 to upgrade oca-addons-repo-template and try again

celm1990 commented:
@pedrobaeza @yajo 
I updated code, could please enable github actions, on my repo test are execute and green
https://github.com/celm1990/doodba-copier-template/actions/runs/6907131396

celm1990 commented:
@pedrobaeza @yajo 
Can you please help me, I run test on my repo but for `V14.0` the tests never finished 
When I removed version 14.0 from the CI, now all the tests are completing successfully. I don't understand why
![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/5db8a297-395d-4b36-a9c9-a37702a71f19)
For version `14.0` I couldn't find relevant information in the logs(I added the  `--verbose` flag)
https://github.com/celm1990/doodba-copier-template/actions/runs/6911268511/job/18805570859

![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/fcf4d115-b21c-4085-ae14-ce0ae8632d18)

celm1990 commented:
@pedrobaeza @yajo 
Help me please, I tried understand test on `doodba-copier-template`, in [this test](https://github.com/Tecnativa/doodba-copier-template/blob/b871898d2fcb9052b7961cec89a63aa968c130cd/tests/test_doodba_qa.py#L49) is used `doodba-qa` to test `pylint` However, as mentioned in another issue, `pylint-odoo` after version `v8.0.6 `is need replace underscore by dashes, Therefore, is it necessary to adapt `doodba-qa` as well, or are these tests obsolete?
Please refer to:
https://github.com/Tecnativa/doodba-qa/blob/fcd49526d9a8deed03d5584912f37e3626d7fba9/insider/pylint#L30

I hope I explained myself well.

yajo commented:
IIRC doodba-qa is archived.

celm1990 commented:
@yajo @pedrobaeza 
I commented on a test related to backup because, for some reason, this test for v14.0 never finishes. Could you enable CI? Additionally, can you review this test if something is causing an issue? I've checked everything and it all seems normal, but I don't understand.

On my repo all test are gree(but without [this commit](https://github.com/Tecnativa/doodba-copier-template/pull/425/commits/8966298e32b99c6aa00fa51da9954c15f5c6283d))
https://github.com/celm1990/doodba-copier-template/actions/runs/6917156814
![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/4f65309a-ecff-4ec1-81ea-979fc042b921)

celm1990 commented:
@yajo @pedrobaeza all tests are OK, CI is green without [this commit](https://github.com/Tecnativa/doodba-copier-template/pull/425/commits/8966298e32b99c6aa00fa51da9954c15f5c6283d), but if these tests are enabled, V14.0 never finishes. I haven't changed the code in this test; everything seems normal, but I don't understand. Any ideas?

Please your help

pedrobaeza commented:
@celm1990 Please remove the last 2 commits and instead, just add one commenting the import of the test in `__init__.py`. We merge this for having the scaffolding available, and later we investigate about the failure.

celm1990 commented:
> @celm1990 Please remove the last 2 commits and instead, just add one commenting the import of the test in `__init__.py`. We merge this for having the scaffolding available, and later we investigate about the failure.

@pedrobaeza I see `__init__.py` and it's empty, where is done import?
![image](https://github.com/Tecnativa/doodba-copier-template/assets/7775116/cc1356dc-20c1-4885-9ceb-550c050c48a7)

pedrobaeza commented:
OK, try to rename the file instead. It seems it imports all the files starting with `test_`.

pedrobaeza commented on .flake8.jinja:
```diff
@@ -1 +1,10 @@
-{%- include "vendor/oca-addons-repo-template/.flake8" -%}
+[flake8]
```
I see that you have hardcoded this. The idea is to use the same linter configuration here that on oca-addons-repo-template, so instead you have to use the new `
.pre-commit-config.yaml` from vendor submodule and remove these ones.

yajo replied:
if you replace for ruff, probably this file will not be needed anymore.

pedrobaeza replied:
Yes, that's the idea. In fact, the `.pre-commit-config.yaml` from oca-addons-repo-template has changed to ruff (not the one being templated):

https://github.com/OCA/oca-addons-repo-template/blob/8ad97a63fde3efe033fb262a6fb73d08a897e73d/.pre-commit-config.yaml#L24

pedrobaeza commented on .pre-commit-config.yaml.jinja:
Replace here black, isort and flake8 by ruff the same as it has been done in oca-addons-repo-template, and I think it can be done for all versions with Python 3 (11+).

celm1990 replied:
@pedrobaeza done, please enable CI

pedrobaeza reviewed (approved):
Thanks for the effort.
